### PR TITLE
use loop kernel for min, max, prod, log_softmax to allow larger reduction size

### PR DESCRIPTION
**Pull request by iclementine**
### PR Category

Bug Fix

### Description

Use loop kernel for min, max, prod, log_softmax to allow larger reduction size.

We leave persistent/loop(stream)/split algorithms and better strategy to support different layouts for further optimization.

### Progress

- [x] Change is properly reviewed (1 reviewer required, 2 recommended).
- [x] Change is responded to an issue.
- [x] Change is fully covered by a UT.

### CODE REVIEW

File: src/flag_gems/ops/log_softmax.py [hunk 1]
```diff
@@ -24,19 +24,8 @@ def heur_num_warps(args):
...
-    ],
-)
-@triton.heuristics(
-    {
-        "BLOCK_N": heur_block_n,
-        "num_warps": heur_num_warps,
-    }
-)
+@triton.autotune(configs=runtime.get_triton_config("log_softmax"), key=["M", "N"])
+@triton.heuristics({"num_warps": heur_num_warps})
```
Comment (StrongSpoon) [inline src/flag_gems/ops/log_softmax.py]: delete heur_block_n

File: src/flag_gems/ops/max.py [hunk 2]
```diff
@@ -77,7 +77,9 @@ def max_kernel(
         mask = m_offset[:, None] < M and n_offset[None, :] < N
         inp_ptrs = inp + offset
         inp_vals = tl.load(inp_ptrs, mask=mask, other=-float("inf"))
-        max_value, max_index = tl.max(inp_vals, axis=1, return_indices=True)
+        max_value, max_index = tl.max(
+            inp_vals, axis=1, return_indices=True, return_indices_tie_break_left=True
```
Comment (StrongSpoon) [inline src/flag_gems/ops/max.py]: why explicitly specify return_indices_tie_break_left?

**Review by StrongSpoon — COMMENTED** [review]
done

File: src/flag_gems/ops/max.py (hunk 2 above)
Comment (iclementine) [reply src/flag_gems/ops/max.py]: deleted

File: src/flag_gems/ops/log_softmax.py (hunk 1 above)
Comment (iclementine) [reply src/flag_gems/ops/log_softmax.py]: done

**Review by StrongSpoon — APPROVED** [review]
lgtm

**Pull request merged**